plugins/bios/fu-plugin-bios.c

```diff
@@ -79,22 +44,26 @@ fu_plugin_coldplug (FuPlugin *plugin, GError **error)
 #if defined(__x86_64__) || defined(__i386__)
 	g_autoptr(GError) error_local = NULL;
 	if (!fu_efivar_supported (&error_local)) {
-		const gchar *reason = "Firmware can not be updated in legacy BIOS mode, switch to UEFI mode";
-		g_warning ("%s", error_local->message);
-		return fu_plugin_bios_create_dummy (plugin, reason, error);
+		fu_plugin_set_mode_text (plugin,
+					 "Firmware can not be updated in "
+					 "legacy BIOS mode, switch to UEFI mode");
+		fu_plugin_set_mode (plugin, FU_PLUGIN_MODE_RT_DEGRADED);
```

Comment:
So if the plugin is runtime degraded mode, there are no devices to set this string to anymore.

Reply:
More to come, but yes.